### PR TITLE
[ci] Add configuration entries for Eiger

### PR DESCRIPTION
**Pull request by jgphpc**

### CODE REVIEW

File: config/cscs.py [hunk 1]
```diff
@@ -627,6 +627,101 @@
...
+            'hostnames': [
+                'eiger'
+            ],
+            'modules_system': 'lmod',
+            'resourcesdir': '/apps/common/UES/reframe/resources',
+            'partitions': [
+                {
+                    'name': 'login',
+                    'scheduler': 'local',
+                    'modules': [],
```
Comment (vkarak) [inline config/cscs.py:642]: Remove this.

File: config/cscs.py [hunk 2]
```diff
@@ -627,6 +627,101 @@
...
+                    'descr': 'Login nodes',
+                    'max_jobs': 4,
+                    'launcher': 'local'
+                },
+                {
+                    'name': 'mc_lowmem',
+                    'descr': 'Multicore nodes (AMD EPYC 7742, 256GB/cn)',
+                    'scheduler': 'slurm',
+                    'access': [
+                        '--exclude=nid00[1512-1703,1768-1927]'
```
Comment (vkarak) [inline config/cscs.py:658]: Can you add a `FIXME`  to remember updating this line?

File: config/cscs.py [hunk 3]
```diff
@@ -627,6 +627,101 @@
...
+                        # f'--constraint=mc',
```
Comment (vkarak) [inline config/cscs.py:659]: Remove this.

File: config/cscs.py [hunk 4]
```diff
@@ -627,6 +627,101 @@
...
+                        'builtin',
+                        'PrgEnv-cray',
+                        'PrgEnv-gnu',
+                        'PrgEnv-aocc'
+                    ],
+                    'max_jobs': 100,
+                    'resources': [
+                        {
+                            'name': 'bcast',
+                            'options': ['--bcast={path}']
```
Comment (vkarak) [inline config/cscs.py:671]: Will we often need this?

File: config/cscs.py [hunk 5]
```diff
@@ -627,6 +627,101 @@
...
+                        },
+                        {
+                            'name': 'switches',
+                            'options': [
+                                '--switches={num_switches}'
+                            ]
+                        },
+                        {
+                            'name': 'mem-per-cpu',
```
Comment (vkarak) [inline config/cscs.py:680]: Will we often need this?

File: config/cscs.py [hunk 6]
```diff
@@ -627,6 +627,101 @@
...
+                        },
+                    ],
+                    'launcher': 'srun'
+                },
+                {
+                    'name': 'mc_himem',
+                    'descr': 'Multicore nodes (AMD EPYC 7742, 512GB/cn)',
+                    'scheduler': 'slurm',
+                    'access': [
+                        '--exclude=nid00[1000-1003,1006-1191,1256-1415]'
```
Comment (vkarak) [inline config/cscs.py:693]: Same comment as above here.

**Review by vkarak — CHANGES_REQUESTED** [review]

**Comment by jgphpc** [comment]
You're right, I can't comment inline if I am not a reviewer 
My comment was:
for large job, we point users to: https://user.cscs.ch/access/running/#large-jobs

**Comment by jgphpc** [comment]
We have: `eiger:mc_lowmem` and `eiger:mc_himem`
I could also add the full partition: `eiger:mc` ?

**Comment by jgphpc** [comment]
Looks like the constraint will not be set. The users will have to use the `--mem=` slurm flag.

**Comment by vkarak** [comment]
> Looks like the constraint will not be set. The users will have to use the `--mem=` slurm flag.

So the way we have set up the configuration now is not correct. In such case, I would suggest to have a single partition and add the memory as a resource.

File: config/cscs.py [hunk 7]
```diff
@@ -627,6 +627,108 @@
...
+                    'max_jobs': 100,
+                    'resources': [
+                        {
+                            'name': 'switches',
+                            'options': [
+                                '--switches={num_switches}'
+                            ]
+                        },
+                        {
+                            'name': 'mem-per-node',
```
Comment (vkarak) [inline config/cscs.py:671]:
```suggestion
                            'name': 'memory',
```

File: config/cscs.py [hunk 8]
```diff
@@ -627,6 +627,108 @@
...
```
Comment (vkarak) [inline config/cscs.py:722]:
```suggestion
                            'name': 'memory',
```

File: config/cscs.py [hunk 9]
```diff
@@ -773,6 +875,33 @@
             'cxx': 'g++',
             'ftn': 'gfortran'
         },
+        {
+            'name': 'PrgEnv-cray',
+            'target_systems': [
+                'alps', 'eiger', 'rigi', 'pilatus'
```
Comment (vkarak) [inline config/cscs.py:881]: I would only put the systems that we have defined here.

```suggestion
                'eiger', 'pilatus'
```

File: config/cscs.py [hunk 10]
```diff
@@ -773,6 +875,33 @@
...
+            ],
+            'modules': [
+                {'name': 'PrgEnv-cray', 'collection': True}
+            ]
+        },
+        {
+            'name': 'PrgEnv-gnu',
+            'target_systems': [
+                'alps', 'eiger', 'rigi', 'pilatus'
```
Comment (vkarak) [inline config/cscs.py:890]: Similarly here.

**Review by vkarak — CHANGES_REQUESTED** [review]
I have only a couple of minor comments that I could fix myself.

**Comment by jgphpc** [comment]
I want to add `mem-per-cpu` to the config and here is why we need it:

> scontrol show conf

`DefMemPerCPU = 1000`

ok ?

File: config/cscs.py [hunk 11]
```diff
@@ -668,9 +668,9 @@
                             ]
                         },
                         {
-                            'name': 'mem-per-node',
+                            'name': 'memory',
                             'options': [
-                                '--mem={mem_per_node}'
```
Comment (vkarak) [inline config/cscs.py]: This you can still keep it as `mem_per_node`. The name of the resource can have multiple options, that's they should not necessarily be the same.

**Review by vkarak — APPROVED** [review]
lgtm

**Pull request merged**